frontend/public/kubevirt/components/utils/withResources.jsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash-es';
+
+import { inject } from '../../../components/utils';
+
+import { Loader } from '../modals/loader';
+import { showErrors } from './showErrors';
+
+
+/*
+ * Firehose helper
+ */
+export class WithResources extends React.Component {
+
+  constructor (props) {
+    super(props);
+    this.state = WithResources.getDerivedStateFromProps(props);
+    WithResources.checkErrors(this.state.errors, props.dispose);
+  }
+
+  static getDerivedStateFromProps ({ resourceMap, resources, resourceToProps }) {
+    const childrenProps = {};
+    let errors = [];
+    let loaded = true;
+
+    Object.keys(resourceMap).forEach(resourceKey => {
+      const resourceConfig = resourceMap[resourceKey];
+      const resource = _.get(resources, resourceConfig.resource.kind);
+
+      if (resource) {
+        if (resource.loaded) {
+          childrenProps[resourceKey] = resource.data;
+        } else if (resourceConfig.isRequired) {
+          loaded = false;
+        }
+
+        if (resource.loadError) {
+          errors.push(resource.loadError);
+        }
+      }
+      // skip unknown resources (CRD not created in opeshift, etc..)
+    });
+
+    return {
+      childrenProps: {
+        ...childrenProps,
+        ...(resourceToProps ? resourceToProps(childrenProps) : {})
+      },
+      errors,
+      loaded,
+    };
+  }
+
+  static checkErrors (errors, dispose) {
```

Comment:
any reason to have this as `static` method ?

Reply:
I guess it can be a separated function